dev-scripts/openshift_deploy.sh

```diff
@@ -0,0 +1,233 @@
+  DEFAULT_OPENSHIFT_NAMESPACE_URL="${CHE_OPENSHIFT_PROJECT}.$(minishift ip).nip.io"
+  OPENSHIFT_NAMESPACE_URL=${OPENSHIFT_NAMESPACE_URL:-${DEFAULT_OPENSHIFT_NAMESPACE_URL}}
+  DEFAULT_CHE_KEYCLOAK_DISABLED="true"
+  CHE_KEYCLOAK_DISABLED=${CHE_KEYCLOAK_DISABLED:-${DEFAULT_CHE_KEYCLOAK_DISABLED}}
+
+elif [ "${OPENSHIFT_FLAVOR}" == "osio" ]; then
+  # ----------------------
+  # Set osio configuration
+  # ----------------------
+  if [ -z "${OPENSHIFT_ID+x}" ]; then echo "[CHE] **ERROR** Env var OPENSHIFT_ID is unset. You need to set it with you OSIO ID to continue. Aborting"; exit 1; fi
+
+  if [ -z "${OPENSHIFT_TOKEN+x}" ]; then echo "[CHE] **ERROR** Env var OPENSHIFT_TOKEN is unset. You need to set it with you OSO token to continue. To retrieve your token: https://console.starter-us-east-2.openshift.com/console/command-line. Aborting"; exit 1; fi
+  
+  DEFAULT_OPENSHIFT_ENDPOINT="https://api.starter-us-east-2.openshift.com"
+  OPENSHIFT_ENDPOINT=${OPENSHIFT_ENDPOINT:-${DEFAULT_OPENSHIFT_ENDPOINT}}
+  DEFAULT_CHE_OPENSHIFT_PROJECT="${OPENSHIFT_ID}-che"
+  CHE_OPENSHIFT_PROJECT=${CHE_OPENSHIFT_PROJECT:-${DEFAULT_CHE_OPENSHIFT_PROJECT}}
+  DEFAULT_OPENSHIFT_NAMESPACE_URL="${CHE_OPENSHIFT_PROJECT}.8a09.starter-us-east-2.openshiftapps.com"
+  OPENSHIFT_NAMESPACE_URL=${OPENSHIFT_NAMESPACE_URL:-${DEFAULT_OPENSHIFT_NAMESPACE_URL}}
+  DEFAULT_CHE_KEYCLOAK_DISABLED="false"
+  CHE_KEYCLOAK_DISABLED=${CHE_KEYCLOAK_DISABLED:-${DEFAULT_CHE_KEYCLOAK_DISABLED}}
+fi
+
+# ---------------------------------------
+# Verify that we have all env var are set
+# ---------------------------------------
+if ([ -z "${OPENSHIFT_USERNAME+x}" ] || 
+    [ -z "${OPENSHIFT_PASSWORD+x}" ]) && 
+    [ -z "${OPENSHIFT_TOKEN+x}" ]; then echo "[CHE] **ERROR** Env var OPENSHIFT_USERNAME, OPENSHIFT_PASSWORD and OPENSHIFT_TOKEN are unset. You need to set username/password or token to continue. Aborting"; exit 1; fi
+
+if [ -z "${OPENSHIFT_ENDPOINT+x}" ]; then echo "[CHE] **ERROR**Env var OPENSHIFT_ENDPOINT is unset. You need to set it to continue. Aborting"; exit 1; fi
+if [ -z "${OPENSHIFT_NAMESPACE_URL+x}" ]; then echo "[CHE] **ERROR**Env var OPENSHIFT_NAMESPACE_URL is unset. You need to set it to continue. Aborting"; exit 1; fi
+
+# -----------------------------------
+# Logging on to the OpenShift cluster
+# -----------------------------------
+echo -n "[CHE] Logging on using OpenShift endpoint \"${OPENSHIFT_ENDPOINT}\"..."
+if [ -z ${OPENSHIFT_TOKEN+x} ]; then
+  oc login "${OPENSHIFT_ENDPOINT}" -u "${OPENSHIFT_USERNAME}" -p "${OPENSHIFT_PASSWORD}" > /dev/null
+  OPENSHIFT_TOKEN=$(oc whoami -t)
+else
+  oc login "${OPENSHIFT_ENDPOINT}" --token="${OPENSHIFT_TOKEN}"  > /dev/null
+fi
+echo "done!"
+
+# --------------------------
+# Create project (if needed)
+# --------------------------
+DEFAULT_CHE_OPENSHIFT_PROJECT="eclipse-che"
+CHE_OPENSHIFT_PROJECT=${CHE_OPENSHIFT_PROJECT:-${DEFAULT_CHE_OPENSHIFT_PROJECT}}
```

Comment:
This check should happen earlier -- it duplicates some code in the `# Set osio configuration` section, and the `# Set minishift configuration` section depends on `CHE_OPENSHIFT_PROJECT` for `DEFAULT_OPENSHIFT_NAMESPACE_URL`